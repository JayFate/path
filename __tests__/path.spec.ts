import path from "../src/index"

// [
//   'resolve',    'normalize',
//   'isAbsolute', 'join',
//   'relative',   'toNamespacedPath',
//   'dirname',    'basename',
//   'extname',    'format',
//   'parse',      'sep',
//   'delimiter',  'win32',
//   'posix',      '_makeLong'
// ]

// process.chdir(`/Users/11104760/Documents/study/webpack`)

const macBase = `/Users/jayfate/Downloads/path`
const winBase = `D:\\jayfate\\toolkit1`
const sep = `~`.repeat(30)
const str1 = 'a/b/c'
const str2 = 'a\\b\\c'
const node_modules = path.resolve(__dirname, "../node_modules")
const thisFile = path.resolve(__dirname, 'index.ts')

describe('compiler: element transform', () => {
  test('import + resolve component', () => {
    const arr = [
      0 + sep,
      path.resolve(macBase),
      // path.resolve(winBase) 的结果与 working dir 有关
      path.resolve(winBase),
      path.resolve(macBase, "build"),
      path.resolve(winBase, "build"),
      // on linux/macos
      // path.win32 = {} 
      // path.win32.resolve(macBase),
      path.posix.resolve(macBase),
      // path.win32.resolve(macBase, "build"),
      path.posix.resolve(macBase, "build"),
      // path.posix 并不能正确的处理 windows 的路径
      path.posix.resolve(winBase),
      path.posix.resolve(winBase, "build"),
      1 + sep,
      path.normalize('/foo/bar//baz/asdf/quux/..'),
      path.normalize('C:\\temp\\\\foo\\bar\\..\\'),
      // path.win32.normalize('/foo/bar//baz/asdf/quux/..'),
      // path.win32.normalize('C:\\temp\\\\foo\\bar\\..\\'),
      path.posix.normalize('/foo/bar//baz/asdf/quux/..'),
      path.posix.normalize('C:\\temp\\\\foo\\bar\\..\\'),
      2 + sep,
      path.isAbsolute(macBase),
      path.isAbsolute(winBase),
      // path.win32.isAbsolute(macBase),
      path.posix.isAbsolute(winBase),
      path.isAbsolute(str1),
      path.isAbsolute(str2),
      3 + sep,
      path.join(macBase),
      path.join(winBase),
      path.join(macBase, "build"),
      path.join(winBase, "build"),
      // path.win32.join(macBase),
      path.posix.join(winBase),
      // path.win32.join(macBase, "build"),
      path.posix.join(winBase, "build"),
      4 + sep,
      path.relative(__dirname, node_modules),
      // path.win32.relative(__dirname, node_modules),
      path.posix.relative(__dirname, node_modules),
      5 + sep,
      path.toNamespacedPath(macBase),
      path.toNamespacedPath(winBase),
      6 + sep,
      path.dirname(thisFile),
      path.dirname(thisFile),
      7 + sep,
      path.basename(thisFile),
      path.basename(thisFile),
      8 + sep,
      path.extname(thisFile),
      path.extname(thisFile),
      9 + sep,
      path.format({
        root: '/ignored',
        dir: '/home/user/dir',
        base: 'file.txt'
      }),
      path.format({
        root: '/',
        base: 'file.txt',
        ext: 'ignored'
      }),
      path.format({
        root: '/',
        name: 'file',
        ext: '.txt'
      }),
      path.format({
        root: '/',
        name: 'file',
        ext: 'txt'
      }),
      path.format({
        dir: 'C:\\path\\dir',
        base: 'file.txt'
      }),
      10 + sep,
      path.parse('/home/user/dir/file.txt'),
      path.parse('C:\\path\\dir\\file.txt'),
      11 + sep,
      `path.sep: ` + path.sep,
      `path.delimiter: ` + path.delimiter,
      12 + sep,
      `path.win32: ` + Object.keys(path.win32).join(', '),
      `path.posix: ` + Object.keys(path.posix).join(', '),
      // path._makeLong
    ]
    expect(arr).toMatchSnapshot()
  })

})
