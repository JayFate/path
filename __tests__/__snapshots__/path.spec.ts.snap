// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler: element transform import + resolve component 1`] = `
[
  "0~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "/Users/jayfate/Downloads/path",
  "/workspace/path/D:/jayfate/toolkit1",
  "/Users/jayfate/Downloads/path/build",
  "/workspace/path/D:/jayfate/toolkit1/build",
  "/Users/jayfate/Downloads/path",
  "/Users/jayfate/Downloads/path/build",
  "/workspace/path/D:\\jayfate\\toolkit1",
  "/workspace/path/D:\\jayfate\\toolkit1/build",
  "1~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "/foo/bar/baz/asdf",
  "C:/temp/foo/",
  "/foo/bar/baz/asdf",
  "C:\\temp\\\\foo\\bar\\..\\",
  "2~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  true,
  false,
  false,
  false,
  false,
  "3~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "/Users/jayfate/Downloads/path",
  "D:/jayfate/toolkit1",
  "/Users/jayfate/Downloads/path/build",
  "D:/jayfate/toolkit1/build",
  "D:\\jayfate\\toolkit1",
  "D:\\jayfate\\toolkit1/build",
  "4~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "../node_modules",
  "../node_modules",
  "5~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "/Users/jayfate/Downloads/path",
  "D:/jayfate/toolkit1",
  "6~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "/workspace/path/__tests__",
  "/workspace/path/__tests__",
  "7~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "index.ts",
  "index.ts",
  "8~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  ".ts",
  ".ts",
  "9~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "/home/user/dir/file.txt",
  "/file.txt",
  "/file.txt",
  "/filetxt",
  "C:/path/dir/file.txt",
  "10~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  {
    "base": "file.txt",
    "dir": "/home/user/dir",
    "ext": ".txt",
    "name": "file",
    "root": "/",
  },
  {
    "base": "file.txt",
    "dir": "C:/path/dir",
    "ext": ".txt",
    "name": "file",
    "root": "",
  },
  "11~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "path.sep: /",
  "path.delimiter: :",
  "12~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~",
  "path.win32: ",
  "path.posix: resolve, normalize, isAbsolute, join, relative, toNamespacedPath, dirname, basename, extname, format, parse, sep, delimiter, win32, posix, _makeLong",
]
`;
